include/BSMPT/bounce_solution/bounce_solution.h

```diff
@@ -0,0 +1,557 @@
+// Copyright (C) 2024 Lisa Biermann, Margarete Mühlleitner, Rui Santos, João
+// Viana
+//
+// SPDX-FileCopyrightText: 2024 Lisa Biermann, Margarete Mühlleitner, Rui
+// Santos, João Viana
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+#ifndef INCLUDE_BSMPT_BOUNCE_SOLUTION_H_
+#define INCLUDE_BSMPT_BOUNCE_SOLUTION_H_
+
+/**
+ * @file bounce solution calculation
+ */
+
+#include <BSMPT/bounce_solution/action_calculation.h>
+#include <BSMPT/minimum_tracer/minimum_tracer.h> // MinimumTracer
+#include <BSMPT/utility/spline.h>
+#include <Eigen/Dense>
+#include <algorithm>             // std::max
+#include <gsl/gsl_deriv.h>       // numerical derivative
+#include <gsl/gsl_integration.h> // numerical integration
+
+namespace BSMPT
+{
+
+/**
+ * struct to store result and error
+ */
+struct resultErrorPair
+{
+  double result;
+  double error;
+};
+
+class BounceSolution
```

Comment:
Maybe add a short description of the class?

Reply:
I added a short description.